pkg/build/api/types.go

```diff
@@ -397,6 +397,9 @@ const (
 	// availability of a new version of an image
 	ImageChangeBuildTriggerType           BuildTriggerType = "ImageChange"
 	ImageChangeBuildTriggerTypeDeprecated BuildTriggerType = "imageChange"
+
+	// ConfigChangeBuildTriggerType will trigger a build on an initial build config creation
+	ConfigChangeBuildTriggerType BuildTriggerType = "ConfigChange"
```

Comment:
i would assume so and yes that feels a little ugly but i don't see a better alternative.